src/operator/contrib/krprod.h

```diff
@@ -0,0 +1,202 @@
+/*!
+ *  Copyright (c) 2017 by Contributors
+ *  \file krprod.h
+ *  \brief Core function for Khatri-Rao product
+ *  \author Jencir Lee
+ */
+#ifndef MXNET_OPERATOR_CONTRIB_KRPROD_H_
+#define MXNET_OPERATOR_CONTRIB_KRPROD_H_
+#include <vector>
+#include "mshadow/tensor.h"
+#include "mxnet/c_lapack_api.h"
+
+namespace mxnet {
+namespace op {
+
+using namespace mshadow;
+using namespace mshadow::expr;
+
+/*!
+ * \brief Computes row-wise Kronecker product
+ *
+ * Given input matrices, this function computes the Kronecker product
+ * row-wise. E.g. if the input matrices  are of shape (3, 2), (3, 4),
+ * (3, 5), the result matrix will be of shape (3, 2 * 4 * 5), which is
+ * (3, 40).
+ *
+ * \param out result matrix
+ * \param ts_arr vector of input matrices
+ */
+template <typename DType>
+inline void row_wise_kronecker
+  (Tensor<cpu, 2, DType> out,
+  const std::vector<Tensor<cpu, 2, DType> > &ts_arr) {
+  // If no input matrix, return all-one vector
+  if (ts_arr.empty()) {
+    CHECK_EQ(1, out.size(1)) << "The output matrix must have single column.";
+    out = 1;
+    return;
+  }
+
+  // Check all input and output matrices have the same number of rows
+  // and the output matrix has the right number of columns
+  int nrows = static_cast<int>(out.size(0));
+  int kronecker_length = 1;
```

Comment:
These lines are within the function `row_wise_kronecker()`, so I named it `kronecker_length`.